### PR TITLE
Fix remove from favorite fragment

### PR DESCRIPTION
**Pull request by sirambd**
Fix issue #417 
- [x] remove on favoriteFragment
- [x] Show empty view if needed

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/fileList/FileAdapter.kt [hunk 1]
```diff
@@ -80,7 +81,8 @@ open class FileAdapter(
             }
 
             override fun onItemRangeRemoved(positionStart: Int, itemCount: Int) {
-                if (viewHolderType == DisplayType.LIST && fileList.isNotEmpty()) {
+                if (fileList.isEmpty()) onEmptyList?.invoke()
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/FileAdapter.kt:84]: Plz add Bracket

**Review by JorisBodin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/fileList/FavoritesFragment.kt [hunk 2]
```diff
@@ -41,15 +41,21 @@ class FavoritesFragment : FileListFragment() {
...
+
+        fileAdapter.apply {
+            onEmptyList = { changeNoFilesLayoutVisibility(hideFileList = true, changeControlsVisibility = false) }
+
+            onFileClicked = { file ->
+                if (file.isFolder()) {
+                    fileListViewModel.cancelDownloadFiles()
+                    safeNavigate(FavoritesFragmentDirections.actionFavoritesFragmentSelf(file.id, file.name))
+                } else {
+                    val fileList = fileAdapter.getFileObjectsList(mainViewModel.realm)
```
[the remark this answers is not in the document]
Comment (sirambd) [reply app/src/main/java/com/infomaniak/drive/ui/fileList/FavoritesFragment.kt:53]: 👍🏽

File: app/src/main/java/com/infomaniak/drive/ui/fileList/FavoritesFragment.kt [hunk 3]
```diff
@@ -70,10 +76,11 @@ class FavoritesFragment : FileListFragment() {
                 it?.let { result ->
                     if (fileAdapter.itemCount == 0 || result.page == 1 || isNewSort) {
                         val realmFiles = FileController.getRealmLiveFiles(
+                            isFavorite = true,
                             order = fileListViewModel.sortType,
                             parentId = FileController.FAVORITES_FILE_ID,
                             realm = mainViewModel.realm,
-                            withVisibilitySort = false
+                            withVisibilitySort = false,
```
[the remark this answers is not in the document]
Comment (sirambd) [reply app/src/main/java/com/infomaniak/drive/ui/fileList/FavoritesFragment.kt:83]: For a quick sort, but it's not important

[the remark this answers is not in the document]
Comment (sirambd) [reply app/src/main/java/com/infomaniak/drive/ui/fileList/FavoritesFragment.kt:83]: Removed

File: app/src/main/java/com/infomaniak/drive/ui/fileList/FileAdapter.kt (hunk 1 above)
Comment (sirambd) [reply app/src/main/java/com/infomaniak/drive/ui/fileList/FileAdapter.kt:84]: - 🤔 why not, 
 - fixed

**Review by JorisBodin — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**